### ISSUE TITLE
Recipe: kimi-k2.5-dpo and kimi-k2.5-grpo (part of #275)

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#851]
## What

Add two recipes to `src/soup_cli/recipes/catalog.py`:

- `kimi-k2.5-dpo`
- `kimi-k2.5-grpo`

`moonshotai/Kimi-K2.5` ships SFT only (`kimi-k2.5-sft`, v0.71.24). The next model in the line, Kimi-K2.6, already has all three (`-sft`, `-grpo` from #614, `-dpo` from #664), so the templates are one release away and on the same architecture. One PR for both or just one is fine — **comment here naming which** before you start.

## Base repo

`moonshotai/Kimi-K2.5` — checked on 2026-09-11 with `huggingface_hub.model_info(..., token=False)`: resolves, not gated. Re-check it the same way; an HTTP status is not evidence (#661).

The same check reports `model_type: kimi_k25` and pipeline tag `image-text-to-text` for **both** K2.5 and K2.6. None of the three existing K2.6 recipes, nor `kimi-k2.5-sft`, declares `modality:`, so all of them resolve to `text`. Copying the K2.6 templates as they are keeps the family consistent; if you think the family should pin text-only intent explicitly (as #427 did for Qwen3.5), raise it as a separate issue rather than changing it in this PR.

## What to copy, and from where

The K2.5 SFT sibling and the K2.6 recipes share the model block: `max_length: 8192`, `batch_size: 1`, `gradient_accumulation_steps: 16`, `quantization: 4bit`, `moe_lora: true`, `gradient_checkpointing: true`.

| recipe | template | fields that change vs `kimi-k2.5-sft` |
|---|---|---|
| `kimi-k2.5-dpo` | `kimi-k2.6-dpo` | `task: dpo`, `data.train: ./data/preference_train.jsonl`, `data.format: dpo`, `lr: 5e-6`, `dpo_beta: 0.1` (keeps `epochs: 1`, LoRA r=32) |
| `kimi-k2.5-grpo` | `kimi-k2.6-grpo` | `task: grpo`, `data.train: ./data/reasoning_train.jsonl`, `epochs: 3`, LoRA r=16 / alpha=32, `grpo_beta: 0.1`, `num_generations: 4`, `reward_fn: accuracy` |

Counts from the catalog: all 14 DPO recipes use `lr: 5e-6` and `dpo_beta: 0.1`, and the only two with `epochs: 1` are the giant-MoE ones (`glm-5.1-dpo`, `kimi-k2.6-dpo`). All 4 MoE GRPO recipes use `lr: 1e-5`, `batch_size: 1`, `gradient_accumulation_steps: 16`, `gradient_checkpointing: true`. Note `kimi-k2.6-grpo` drops `moe_aux_loss_coeff` while the DPO recipe keeps it; since `0.01` is the schema default that is not a behavioural difference, but mirror the template you copy.

YAML assembled this way parses through `load_config_from_string` on current `main` for both tasks. That proves the schema accepts it, nothing more.

## The ratchets that must pass

- [ ] `RecipeMeta(...)` entries in `catalog.py`, tags/description following `kimi-k2.5-sft` (Modified MIT, ~1T / 32B active, multi-node note).
- [ ] Model id pinned on **two independent surfaces** (`RecipeMeta.model` and YAML `base:`), each by its own test — mirror `TestDeepSeekV4FlashDpoRecipe`, `tests/test_recipes.py:503`. `Kimi-K2.5` vs `Kimi-K2.6` is a one-character difference, which is exactly what copying from a K2.6 template will get wrong.
- [ ] Catalog count bumped everywhere (165 today; use the number at rebase time): `catalog.py:51`; `test_catalog_size_is_N` in `tests/test_recipes.py:936`, `tests/test_v07130.py:740`, `tests/test_v07132.py:953` plus the history line near `tests/test_recipes.py:964`; `tests/test_v07124.py:281` and the docstring at `:277`; the doc sites in `tests/test_recipe_count_is_synced.py:83-96`.
- [ ] `python scripts/generate_recipe_snapshot.py` run last; additive diff only.
- [ ] Changelog fragment `changelog.d/0.74.0/<PR>.added.md`.
- [ ] `soup recipes show` / `use` run live; paste the output.

## A note on mutations

`dpo_beta: 0.1`, `grpo_beta: 0.1`, `num_generations: 4`, `reward_fn: accuracy`, `epochs: 3` and `moe_aux_loss_coeff: 0.01` equal their schema defaults; deleting them is an equivalent mutation. Report them as survivors.

## Honest limits

A ~1T MoE is not trainable for this PR and nobody expects it. The values are inherited from the K2.6 templates, not tuned for K2.5; say so in the PR.

**Comment by Areen-09** [comment]
Hi @MakazhanAlpamys , I’d like to take this up.

**Comment by MakazhanAlpamys** [comment]
@Areen-09 — yours, welcome. The issue asks for one thing before you start: say here whether you are doing **both recipes in one PR or just one**. Either is fine; one PR for both is less churn.

The two places a recipe PR like this usually goes wrong, so you can check them before review does:

1. **`Kimi-K2.5` vs `Kimi-K2.6` is a one-character difference**, and you will be copying from the K2.6 templates. Pin the model id on both surfaces — `RecipeMeta.model` and the YAML `base:` — each with its own test, mirroring `TestDeepSeekV4FlashDpoRecipe` in `tests/test_recipes.py`. Then change one of them to `K2.6` locally and confirm the test goes red.
2. **The catalog count lives in many places.** The issue lists every site. Bump them all to the number at rebase time, since other recipe PRs may land first. Run `python scripts/generate_recipe_snapshot.py` last, and check the diff is additive only.

Please also paste real `soup recipes show` / `soup recipes use` output in the PR body, and list the schema-default fields as equivalent-mutation survivors, as the issue's mutation note describes.

**Comment by MakazhanAlpamys** [comment]
Done in #872 by @Areen-09, merged as f218a7c — ships in the next release.

Both recipes are byte-identical to their `kimi-k2.6-*` templates apart from `base:`, including the asymmetry this issue called out: the GRPO recipe drops `moe_aux_loss_coeff` and the DPO recipe keeps it, each mirroring the template it was copied from rather than being normalised. Catalog count **167 -> 169**, updated everywhere it is live, with the snapshot fixture regenerated purely additively.

Verified before merging that both new entries are actually covered by the #621 ratchet and not merely present in the fixture: mutating the `dpo_beta` schema default turns 17 tests red and both new recipes are among them.

Neither recipe is trained — K2.5 is a ~1T MoE — so no hyperparameter here is a measured recommendation, which the merge commit says explicitly.

Try it:

```
soup recipes show kimi-k2.5-dpo
soup recipes use kimi-k2.5-grpo
```

Still open under #275 for anyone reading: #845, #846, #847, #849, #850.
